node/sc/mainbridge_test.go

```diff
@@ -116,3 +118,80 @@ func TestMainBridge_basic(t *testing.T) {
 
 	//TODO more test
 }
+
+// TestMainBridge_removePeer tests correct removal of a peer from `MainBridge.peers`.
+func TestMainBridge_removePeer(t *testing.T) {
+	// Create a MainBridge (it may have 0 peers)
+	mBridge := testNewMainBridge(t)
+	defer mBridge.chainDB.Close()
+
+	// Prepare a bridgePeer to be added and removed
+	nodeID := "0x1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d1d"
+	peer := p2p.NewPeer(discover.MustHexID(nodeID), "name", []p2p.Cap{})
+	bridgePeer := mBridge.newPeer(1, peer, &p2p.MsgPipeRW{})
+
+	// Add the bridgePeer
+	mBridge.peers.Register(bridgePeer)
+	peerNum := mBridge.peers.Len()
+
+	// Try to remove a non-registered bridgePeer, and nothing happen
+	mBridge.removePeer("0x11111111")
+	assert.Equal(t, peerNum, mBridge.peers.Len())
+
+	// Remove the registered bridgePeer
+	mBridge.removePeer(bridgePeer.GetID())
+	assert.Equal(t, peerNum-1, mBridge.peers.Len())
+}
+
+// TestMainBridge_handleMsg fails when a bridgePeer fails to read a message or reads a too long message.
+func TestMainBridge_handleMsg(t *testing.T) {
+	// Create a MainBridge
+	mBridge := testNewMainBridge(t)
+	defer mBridge.chainDB.Close()
+
+	// Elements for a bridgePeer
+	key, _ := crypto.GenerateKey()
+	nodeID := discover.PubkeyID(&key.PublicKey)
+	peer := p2p.NewPeer(nodeID, "name", []p2p.Cap{})
+	pipe1, pipe2 := p2p.MsgPipe()
+
+	// bridgePeer will receive a message through rw1
+	bridgePeer := newBridgePeer(testProtocolVersion, peer, pipe1)
+
+	// Case1. Send a valid message and handle it successfully
+	{
+		data := "valid message"
+		go func() {
+			if err := p2p.Send(pipe2, StatusMsg, data); err != nil {
+				t.Fatal(err)
+			}
+		}()
+
+		if err := mBridge.handleMsg(bridgePeer); err != nil {
+			t.Fatal(err)
```

Comment:
Are there any reason between using `t.Fatal` and `assert`?

Reply:
I use `t.Fatal` only when some code doesn't expect any specific value except `nil`. It means that the failure of the code is not a part of the test I intended. 

This kind of convention is just my preference. So, I can change it if the change gives more consistency for our code. How do you think @ehnuje ?

Reply:
I also agree with your point =) Thanks for the explanation.